python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -21,131 +21,95 @@
 #
 """Calculation of brighter-fatter effect correlations and kernels."""
 
-__all__ = ['MakeBrighterFatterKernelTaskConfig',
-           'MakeBrighterFatterKernelTask',
-           'calcBiasCorr']
+__all__ = ['BrighterFatterKernelSolveTask',
+           'BrighterFatterKernelSolveConfig']
 
-import os
-import copy
-from scipy import stats
 import numpy as np
-import matplotlib.pyplot as plt
-# the following import is required for 3d projection
-from mpl_toolkits.mplot3d import axes3d  # noqa: F401
-from matplotlib.backends.backend_pdf import PdfPages
-from dataclasses import dataclass
-
-import lsstDebug
-import lsst.afw.image as afwImage
+
 import lsst.afw.math as afwMath
-import lsst.afw.display as afwDisp
-from lsst.ip.isr import IsrTask
-import lsst.log as lsstLog
 import lsst.pex.config as pexConfig
 import lsst.pipe.base as pipeBase
-from .utils import PairedVisitListTaskRunner, checkExpLengthEqual
-import lsst.daf.persistence.butlerExceptions as butlerExceptions
-from lsst.cp.pipe.ptc.measurePtcGen2Task import (MeasurePhotonTransferCurveTaskConfig,
-                                                 MeasurePhotonTransferCurveTask)
-from lsst.ip.isr import PhotonTransferCurveDataset
+import lsst.pipe.base.connectionTypes as cT
+
+from lsst.ip.isr import (BrighterFatterKernel)
+from .utils import (funcPolynomial, irlsFit)
+from ._lookupStaticCalibration import lookupStaticCalibration
+
+
+class BrighterFatterKernelSolveConnections(pipeBase.PipelineTaskConnections,
+                                           dimensions=("instrument", "exposure", "detector")):
```

Comment:
Gruen et al 2015 and Coulton et al 2017 report a weak chromaticity of the BFE (however, Antilogus' model implies that the electric field distortions happen in the last few microns (<< 100 microns thickness of the CCD), so, according to that, the effect should be achromatic...maybe I should ask Craig too and see what he has measured). In any case, if we wanted to study this at some point, would be able to easily derive a BF kernel per filter? I put this comment here because I'm hoping that this is just a matter of adding another "filter" (or however it is called now) dimension at this point in the code, but I might be wrong.

Reply:
We'd need to add `filter` here, and change the `dataset_type_name` used in the `connections.outputBFK` (to `chromaticBrighterFatterKernel` or `chromaticBfk` depending on how verbose).  This constraint is there because a given `dataset_type` can only have one defined set of dimensions.